cloudbridge/cloud/providers/gce/services.py

```diff
@@ -661,3 +667,196 @@ def routers(self):
+        Returns a volume given its id.
+        """
+        try:
+            response = self.provider.get_gce_resource_data(volume_id)
+            if response:
+                return GCEVolume(self.provider, response)
+        except googleapiclient.errors.HttpError as http_error:
+            # If the volume is not found, the API will raise
+            # googleapiclient.errors.HttpError.
+            cb.log.warning(
+                "googleapiclient.errors.HttpError: {0}".format(http_error))
+        return None
+
+    def find(self, name, limit=None, marker=None):
+        """
+        Searches for a volume by a given list of attributes.
+        """
+        filtr = 'name eq ' + name
+        max_result = limit if limit is not None and limit < 500 else 500
+        response = (self.provider
+                        .gce_compute.disks()
+                        .list(project=self.provider.project_name,
+                              zone=self.provider.default_zone,
+                              filter=filtr,
+                              maxResults=max_result,
+                              pageToken=marker).execute())
+        if 'items' not in response:
+            return []
+        gce_vols = [GCEVolume(self.provider, vol)
+                    for vol in response['items']]
+        return ServerPagedResultList(len(gce_vols) > max_result,
+                                     response.get('nextPageToken'),
+                                     False, data=gce_vols)
+
+    def list(self, limit=None, marker=None):
+        """
+        List all volumes.
+        """
+        # For GCE API, Acceptable values are 0 to 500, inclusive.
+        # (Default: 500).
+        max_result = limit if limit is not None and limit < 500 else 500
+        response = (self.provider
+                        .gce_compute.disks()
+                        .list(project=self.provider.project_name,
+                              zone=self.provider.default_zone,
+                              maxResults=max_result,
+                              pageToken=marker).execute())
+        if 'items' not in response:
+            return []
+        gce_vols = [GCEVolume(self.provider, vol)
+                    for vol in response['items']]
+        return ServerPagedResultList(len(gce_vols) > max_result,
+                                     response.get('nextPageToken'),
+                                     False, data=gce_vols)
+
+    def create(self, name, size, zone, snapshot=None, description=None):
+        """
+        Creates a new volume.
+
+        Argument `name` must be 1-63 characters long, and comply with RFC1035.
+        Specifically, the name must be 1-63 characters long and match the
+        regular expression [a-z]([-a-z0-9]*[a-z0-9])? which means the first
+        character must be a lowercase letter, and all following characters must
+        be a dash, lowercase letter, or digit, except the last character, which
+        cannot be a dash.
```

Comment:
I'm not sure there are any good options for GCE if you want to allow for free-form names and descriptions on Volumes. The Name is going to be restricted by what Bai has documented here:

The name must be 1-63 characters long, and comply with RFC1035.

However labels have the same [restrictions](https://cloud.google.com/compute/docs/labeling-resources#restrictions):

- Keys and values cannot be longer than 63 characters each.
- Keys and values can only contain lowercase letters, numeric characters, underscores, and dashes. International characters are allowed.